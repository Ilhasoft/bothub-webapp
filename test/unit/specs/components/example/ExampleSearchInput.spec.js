import { shallowMount, createLocalVue } from '@vue/test-utils';
import BH from 'bh';
import ExampleSearchInput from '@/components/example/ExampleSearchInput';


const localVue = createLocalVue();
localVue.use(BH);

describe('ExampleSearchInput.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(ExampleSearchInput, {
      localVue,
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('value received a object, expected the object turns to string', () => {
    beforeEach(() => {
      wrapper.setProps({
        value: {
          intent: 'intent_value',
          label: 'label_value',
          text: 'text_value',
          entity: 'entity_value',
        },
      });
    });

    test('expect data toString be String type', () => {
      expect(typeof wrapper.vm.toString).toBe('string');
    });
  });

  describe('check current computed data turns string to Dicty', () => {
    beforeEach(() => {
      wrapper.setData({
        toString: 'intent:intent_value label:label_value',
      });
    });

    test('expect computed Data current be Object', () => {
      expect(typeof wrapper.vm.current).toBe('object');
    });
  });

  describe('check submit method emit value', () => {
    beforeEach(() => {
      wrapper.vm.submit();
    });

    test('emitted to be defined', () => {
      expect(wrapper.emitted('input')).toBeDefined();
    });
  });

  describe('check clearTimeOut method works correctly', () => {
    beforeEach(() => {
      wrapper.vm.setTimeoutId = 1;
      wrapper.vm.clearTimeout();
    });

    test('after clearTimeOut called, SetTimeoutId is null', () => {
      expect(wrapper.vm.setTimeoutId).toBeNull();
    });
  });
});
