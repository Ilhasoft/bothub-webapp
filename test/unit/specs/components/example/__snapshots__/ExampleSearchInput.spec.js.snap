// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleSearchInput.vue renders correctly 1`] = `
<form>
  <bhtext-stub class="example-search-input"></bhtext-stub>
</form>
`;
